### PR TITLE
Add test for is_hacky_ecoverwrites in cephfs pool checks

### PR DESCRIPTION
**Pull request by jcsp**

### CODE REVIEW

**Comment by jcsp** [comment]
@athanatos @liewegas is this a useful thing to have in master for folks working on ec overwrite stuff?  Obviously it's all still disabled by default on clusters until someone sets their enable_experimental_unrecoverable_data_corrupting_features etc

**Comment by jdurgin** [comment]
This is useful - though I'll be removing the 'hacky' part of the flag soon. It would help to get more testing on cephfs (another facet in one of the cephfs suites?) as well. The branch I'm working on is https://github.com/ceph/ceph-ci/commits/wip-ec-overwrites-cleanup

**Comment by jcsp** [comment]
@jdurgin cool, with the testing I think the way we'll go is to update filesystem.py to take a flag for using EC when creating its pools, then add a config option to the ceph.py task.  For actually adding the permutation to the suites, we currently run the fs suite exhaustively, we're sort of about to hit the limit of that, but I think we might need something smart to do subsets of things like EC vs. replicated, while remaining exhaustive on the cephfs permutations (I sent a mail to ceph-devel)

**Comment by jcsp** [comment]
Turns out I accidentally included the functional part of this in https://github.com/ceph/ceph/pull/13581/files, so this PR is just the test now.

**Comment by jcsp** [comment]
retest this please

**Comment by jcsp** [comment]
@tchaikov @jdurgin this looks to be passing based on the jenkins check

File: qa/workunits/cephtool/test.sh [hunk 1]
```diff
@@ -986,6 +986,20 @@ function test_mon_mds()
 
   fail_all_mds $FS_NAME
   ceph fs rm $FS_NAME --yes-i-really-mean-it
+
+  # We should be permitted to use an EC pool with overwrites enabled
+  # as the data pool...
+  ceph osd pool set mds-ec-pool debug_white_box_testing_ec_overwrites true --yes-i-really-mean-it
```
Comment (tchaikov) [inline qa/workunits/cephtool/test.sh]: i think, in long term, maybe we should rely on teuthology to perform the integration tests. and we have `qa/suites/rbd/cli/pool/ec-data-pool.yaml` to enable this feature. but i have no strong opinion on this though.

if @jdurgin is happy, i am fine.

**Pull request merged**